app/uk/gov/hmrc/economiccrimelevyregistration/controllers/RegistrationSubmissionController.scala

```diff
@@ -72,26 +73,31 @@ class RegistrationSubmissionController @Inject() (
     request: AuthorisedRequest[_]
   ): EitherT[Future, ResponseError, CreateEclSubscriptionResponsePayload] =
     for {
-      _              <- registrationValidationService
-                          .validateRegistration(EclRegistrationModel(registration, registrationAdditionalInfo))
-                          .asResponseError
-      now             = Instant.now().truncatedTo(ChronoUnit.SECONDS)
-      response       <- dmsService
-                          .submitToDms(registration.base64EncodedFields.flatMap(_.dmsSubmissionHtml), now)
-                          .asResponseError
-      _               = if (appConfig.amendRegistrationNrsEnabled) {
-                          nrsService.submitToNrs(
-                            registration.base64EncodedFields.flatMap(_.nrsSubmissionHtml),
+      _                <- registrationValidationService
+                            .validateRegistration(EclRegistrationModel(registration, registrationAdditionalInfo))
+                            .asResponseError
+      now               = Instant.now().truncatedTo(ChronoUnit.SECONDS)
+      registrationType <- valueOrError(registration.registrationType, "Registration type")
+      response         <- dmsService
+                            .submitToDms(
+                              registration.base64EncodedFields.flatMap(_.dmsSubmissionHtml),
+                              now,
+                              registrationType
+                            )
+                            .asResponseError
+      _                 = if (appConfig.amendRegistrationNrsEnabled && registration.registrationType.contains(Amendment)) {
+                            nrsService.submitToNrs(
+                              registration.base64EncodedFields.flatMap(_.nrsSubmissionHtml),
+                              response.eclReference,
+                              appConfig.eclAmendRegistrationNotableEvent
+                            )
+                          }
+      additionalInfo   <- valueOrError(registrationAdditionalInfo, "Registration additional info")
+      _                 = auditService.successfulSubscriptionAndEnrolment(
```

Comment:
Same here. See above.

Reply:
This is just an audit service call, do we want to fail the whole process if this fails?